src/Uno.UI/UI/LayoutHelper.cs

```diff
@@ -498,19 +498,20 @@ internal static double AspectRatio(this CoreGraphics.CGSize size)
 #endif
 
 		[Pure]
-		internal static Rect GetBoundsRectRelativeTo(this UIElement element, UIElement relativeTo)
+		internal static Rect GetBoundsRectRelativeTo(this FrameworkElement element, FrameworkElement relativeTo)
 		{
 			var elementToTarget = element.TransformToVisual(relativeTo);
-			var elementRect = new Rect(default, element.RenderSize);
+			// Use ActualWidth/ActualHeight which may differ from RenderSize in some cases (notably, TextBlock)
+			var elementRect = new Rect(0, 0, element.ActualWidth, element.ActualHeight);
```

Comment:
Do we change TextBlock's ActualWidth/Height as well ?

Reply:
We do! :) https://github.com/unoplatform/uno/pull/2269